### DISCUSSION TITLE
Affine transformation not giving expected output

### DISCUSSION THREAD

**Discussion by Aveturi13** [Project-MONAI/tutorials#1239] [Q&A]
Hi,

I have a question regarding the Affine transform which I'm using for a project on medical image registration. I am using the MEDNIST dataset where I load a new image (calling it the fixed image) and then I create an affinely transformed version of this image (calling it the moving image). For a simple affine transform, I just try scaling, however, the Affine method doesn't give the expected result. 

`
train_transforms = Compose(
    [
    LoadImageD(keys=["fixed_hand", "moving_hand"]),
    EnsureChannelFirstD(keys=["fixed_hand", "moving_hand"]),
    ScaleIntensityRanged(keys=["fixed_hand", "moving_hand"],
                            a_min=0., a_max=255., b_min=0.0, b_max=1.0, clip=True,),
    AffineD(keys=['moving_hand'], scale_params=(2, 2)),
    ToTensorD(keys=["fixed_hand", "moving_hand"]),
    ]
)
`
In the above code snippet, I expect the moving image to be fixed image upscaled by a factor of 2, but instead, I am getting the opposite:

<img width="1085" alt="Screenshot 2023-03-03 at 12 24 37 AM" src="https://user-images.githubusercontent.com/44966149/222657697-3321f37e-b28b-4ab8-afef-4219e812ccc3.png">

I also noticed this pattern with other parameters e.g. in translation, I set the parameters as 10 for both the x and y direction, but the image moves 10 pixels left and up, rather than right and up. I'm not sure why this is happening. Could someone please help me with this? Thank you so much!

**Comment by wyli** [comment]
The affine transform is transforming the output space to the input space to fetch input values, following the scipy ndimage convention https://docs.scipy.org/doc/scipy/reference/generated/scipy.ndimage.affine_transform.html

**Reply by Aveturi13** [reply]
Thanks a lot for your reply! This is helpful :)
